pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,22 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int = 1
```

Comment:
This variable will be overwritten below so I don't think you need to initialize it to 1?

Reply:
Just wanted to keep a safe default, if the env variable is not set. But I see your point, we can do this in the init().